File: challenges/isSumOfConsecutive/java/IsSumOfConsecutive.java
import java.util.stream.IntStream;

public class IsSumOfConsecutive {
	public static void main(String[] args) {
		int[] input0 = new int[] {9, 8, 1, 42, 110, 45, 128, 55, 400, 256};
		boolean[] expectedOutput = new boolean[] {true, false, false, true, true, true, false, true, true, false};
		assert input0.length == expectedOutput.length : String.format("# input0 = %d, # expectedOutput = %d", input0.length, expectedOutput.length);
		IntStream.range(0, expectedOutput.length).forEach(i -> {
			boolean actualOutput = isSumOfConsecutive(input0[i]);
			assert actualOutput == expectedOutput[i] : String.format("isSumOfConsecutive(%d) returned %b, but expected %b", input0[i], actualOutput, expectedOutput[i]);
		});
		System.out.println(String.format("PASSES %d out of %d tests", expectedOutput.length, expectedOutput.length));
	}

	public static boolean isSumOfConsecutive(int n) {
		int m = (int)Math.ceil(n/2.0);
		for (int i = 1; i < m; i++)
			for (int j = i, k = 0; j < m; j++) {
				k += j;
				if (k == n)
					return true;
				else if (k > n)
					break;
			}
		return false;
	}
}

File: challenges/isSumOfConsecutive/scala/isSumOfConsecutive.scala
object isSumOfConsecutive {
	def main(args: Array[String]): Unit = {
		val input0: Array[Int] = Array(9, 8, 1, 42, 110, 45, 128, 55, 400, 256)
		val expectedOutput: Array[Boolean] = Array(true, false, false, true, true, true, false, true, true, false)
		assert(input0.length == expectedOutput.length, s"# input0 = ${input0.length}, # expectedOutput = ${expectedOutput.length}")
		for (i <- 0 until expectedOutput.length) {
			val actualOutput: Boolean = isSumOfConsecutive(input0(i))
			assert(actualOutput == expectedOutput(i), s"""isSumOfConsecutive(${input0(i)}) returned ${actualOutput}, but expected ${expectedOutput(i)}""")
		}
		println(s"PASSES ${expectedOutput.length} out of ${expectedOutput.length} tests")
	}

	def isSumOfConsecutive(n: Int): Boolean = {
		val m: Int = math.ceil(n/2.0).toInt
		for (i <- 0 to m) {
			var k: Int = 0
			var j: Int = i;
			while (j < m && k < n) {
				k += j
				if (k == n)
					return true
				j += 1
			}
		}
		false
	}
}
